src/data-workspace/indicators-table-body/use-indicator-value.js

```diff
@@ -57,6 +58,7 @@ export const useIndicatorValue = ({
                 numerator,
                 factor,
                 formState: form.getState(),
+                decimals,
```

Comment:
I know this wasn't added in this PR. However I'm pretty sure this `useMemo`, really should've been an `useEffect`. Because (at least from react18), we shouldn't read or write to the ref during render.

Reply:
> Because (at least from react18, I don't think it was a "rule" before react18), we shouldn't read or write to the ref during render.

Agree that this could use a rewrite, but I was under the impression that the explicit rules in react18 were due to the concurrent rendering in react 18, which could cause the ref to fall out of sync?

I'm going to leave this code as is for now as I would prefer to work on other things rather than refactoring this. I don't really want to open a ticket either as it will end up being low priority. Maybe we should have a would-be-nice-to-do list for data entry app 😀

Reply:
> but I was under the impression that the explicit rules in react18 were due to the concurrent rendering in react 18, which could cause the ref to fall out of sync? 

Yeah, I believe so as well. So it shouldn't _really_ matter for us, but it could potentially be a cause of bugs in the future. But even in react 18, concurrent rendering is opt-in. 

> I'm going to leave this code as is for now as I would prefer to work on other things rather than refactoring this. I don't really want to open a ticket either as it will end up being low priority.

Agree, and it's not trivial to rework this either, especially while trying to follow all react rules and "best practices".